File: sample-videochat-webrtc/sample-videochat-webrtc/Components/Call/CallHelper/CallKitInfo.m
//
//  CallKitInfo.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 11.09.2020.
//  Copyright © 2020 QuickBlox Team. All rights reserved.
//

#import "CallKitInfo.h"

@interface CallKitInfo ()
//MARK: - Properties
@property (nonatomic, strong) NSString *sessionId;
@property (nonatomic, strong) NSUUID *uuid;
@property (nonatomic, assign) BOOL hasVideo;

@end

@implementation CallKitInfo
//MARK: - Life Cycle
- (id)initWithSessionId:(NSString *)sessionId hasVideo:(BOOL)hasVideo {
    self = [super init];
    if (self) {
        self.sessionId = sessionId;
        self.uuid = sessionId.length ? [[NSUUID alloc] initWithUUIDString: sessionId] : NSUUID.UUID;
        self.hasVideo = hasVideo;
    }
    return self;
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/Calls/CallHelper/CallKitInfo.swift
//
//  CallKitInfo.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 30.03.2021.
//  Copyright © 2021 QuickBlox. All rights reserved.
//

import Foundation

struct CallKitInfo {
    //MARK: - Properties
    var sessionID = ""
    var hasVideo: Bool
    var uuid: UUID
    
    //MARK: - Life Cycle
    init(sessionID: String, hasVideo: Bool) {
        self.sessionID = sessionID
        self.hasVideo = hasVideo
        if sessionID.isEmpty == false, let callUuid = UUID(uuidString: sessionID) {
            self.uuid = callUuid
            return
        }
        self.uuid = UUID()
    }
}
